core/src/main/scala/org/apache/spark/scheduler/CoarseCookSchedulerBackend.scala

```diff
@@ -349,80 +367,93 @@ class CoarseCookSchedulerBackend(
         }
       }
     }
-  }
 
-  /**
-   * Kill the given list of executors through the cluster manager.
-   *
-   * @return whether the kill request is acknowledged.
-   */
   override def doKillExecutors(executorIds: Seq[String]): Future[Boolean] =
     Future.successful {
-      val jobIdsToKill = executorIds.flatMap(executorIdToJobId.get)
-      jobClient.abort(jobIdsToKill.asJava)
-      jobIdsToKill.foreach(abortedJobIds.add)
-      true
+      // No need to adjust `executorLimitOption` since the AllocationManager already communicated
+      // the desired limit through a call to `doRequestTotalExecutors`.
+      // See [[o.a.s.scheduler.cluster.CoarseGrainedSchedulerBackend.killExecutors]]
+      val unknownExecutorIds =
+        executorIds.filterNot(executorIdToJobUUID.contains)
+      if (unknownExecutorIds.nonEmpty) {
+        logWarning(
+          s"Ignore killing unknown executor ids ${unknownExecutorIds.mkString(",")}")
+      }
+
+      val jobUUIDsToAbort = executorIds.flatMap(executorIdToJobUUID.get)
+
+      if (jobUUIDsToAbort.nonEmpty) {
+        val abortJobMessage = s"abort jobs ${jobUUIDsToAbort.mkString(",")}" +
+          s" when calling doKillExecutors(${executorIds.mkString(",")})"
+
+        Try[Unit](jobClient.abort(jobUUIDsToAbort.asJava)) match {
+          case Failure(e) =>
+            logWarning(s"Failed to $abortJobMessage", e)
+            false
+          case Success(_) =>
+            jobUUIDsToAbort.foreach { jobUUID =>
```

Comment:
I'm a little worried that this class contains a bunch of other logic, plus the management of these UUID sets.

Can we factor out a separate class that only contains the UUID sets and has methods for submitting and aborting jobs, and whose only purpose is to submit those requests and maintain the corresponding UUID sets?  I think that will contain the complexity of keeping them in sync in one place and be easier to reason about.